plugins/mapper-annotated-text/src/main/java/org/elasticsearch/index/mapper/annotatedtext/AnnotatedTextFieldMapper.java

```diff
@@ -272,9 +271,16 @@ public AnnotationToken getAnnotation(int index) {
     // The class takes markedup format field values and returns plain text versions.
     // When asked to tokenize plain-text versions by the highlighter it tokenizes the
     // original markup form in order to inject annotations.
+    // Unlike other Analyzers, which tend to be single-instance, this class has
+    // instances created per search request and field being highlighted. This allows us to keep
+    // state about the annotations being processed and pass them into token streams
+    // being highlighted.
```

Comment:
Might be worth changing this to javadoc, and stressing that you *must* create new instances per search because it's not thread-safe.

Reply:
Will do. Thanks for the review!